webapp/channels/src/components/admin_console/password_settings.tsx

```diff
@@ -13,8 +13,50 @@ import Setting from './setting';
 import SettingsGroup from './settings_group.jsx';
 import TextSetting from './text_setting';
 
-export default class PasswordSettings extends AdminSettings {
-    constructor(props) {
+type Props = {
+    config: {
+        PasswordSettings: {
+            MinimumLength: number;
+            Lowercase: boolean;
+            Number: boolean;
+            Uppercase: boolean;
+            Symbol: boolean;
+        };
+        ServiceSettings: {
+            MaximumLoginAttempts: number;
+        };
+        ExperimentalSettings?: {
+            RestrictSystemAdmin: boolean;
+        };
+    };
+    isDisabled: boolean;
+}
+
+type BaseState = {
+    saveNeeded: boolean;
+    saving: boolean;
+    serverError: string;
+    errorTooltip: boolean;
+};
```

Comment:
Base state is actually from
`import AdminSettings, {BaseProps, BaseState} from 'components/admin_console/admin_settings';`